packages/runtime-dom/__tests__/customElement.spec.ts

```diff
@@ -471,6 +471,26 @@ describe('defineCustomElement', () => {
       container.appendChild(e)
       expect(e.shadowRoot!.innerHTML).toBe('<div></div>')
     })
+
+    // https://github.com/vuejs/core/issues/12964
+    // Disabled because of missing support for `delegatesFocus` in jsdom
+    // https://github.com/jsdom/jsdom/issues/3418
+    // eslint-disable-next-line vitest/no-disabled-tests
+    test.skip('shadowRoot should be initialized with delegatesFocus', () => {
```

Comment:
Maybe we can use vitest's browser mode for testing, but it's an experimental feature for now.
https://vitest.dev/guide/browser/#configuration